### PR TITLE
Add support for Freeze Periods

### LINKED ISSUE #1227: Support the Freeze Period API
As far as I can see the API managing [freeze periods](https://gitlab.com/gitlab-org/gitlab/-/blob/master/doc/api/freeze_periods.md) is missing.

Is this something you plan to add near-term or would you be interested in a PR for this?

### PR DESCRIPTION
**Pull request by nlochschmidt**
Reference https://gitlab.com/gitlab-org/gitlab/-/blob/master/doc/api/freeze_periods.md

@jdalrymple I have two additional remarks:
1. I was a bit unclear as to whether I should also write an integration test. There seem to be not that many and I couldn't get the existing ones to run.
2. The [contribution guidelines say](https://github.com/jdalrymple/gitbeaker/blob/master/CONTRIBUTING.md#pull-request-process) that I should also bump the version number but that process seems to be done by you, correct?

Closes #1227

### CODE REVIEW

**Comment by jdalrymple** [comment]
Awesome!

1. Yea i should update the docs, the integration tests only work with a running instance of Gitlab. The unit tests can be run locally without issue though
2. Exactly, its done automatically. I also need to update these docs!

**Comment by nlochschmidt** [comment]
I noticed a wrong ordering in a couple of imports and in the Readme. Should be fixed now.

**Comment by nlochschmidt** [comment]
This is a bit off topic:
> 1. Yea i should update the docs, the integration tests only work with a running instance of Gitlab. The unit tests can be run locally without issue though
FYI I used the docker-compose.yml in the `scripts` folde,r however there is no yarn script `test-with-token` as far as I could see. `yarn test:integration` failed because of `401 (Unauthorized)`. That's when I saw that there are not a lot of integration tests anyways and I gave up. 😅

**Comment by jdalrymple** [comment]
Yea dont worry about the integration tests. If you realllllly wanted to write one, you can avoid the local setup, write the code, push it to the branch and see the build on the Gitlab CI, but its not a big deal so no stress. Ill have to update those docs when I get the chance

**Comment by nlochschmidt** [comment]
I am good 😉

File: README.md [hunk 1]
```diff
@@ -753,6 +755,7 @@ This started as a fork from [node-gitlab-legacy](https://github.com/node-gitlab/
 - [Claudio Vellage](https://github.com/jdalrymple/LRH539)
 - [Seb0uil](https://github.com/seb0uil)
 - [Dylan Taylor](https://github.com/dylanmtaylor)
+- [Niklas Lochschmidt](https://github.com/nlochschmidt)
```
Comment (jdalrymple) [inline README.md:758]: Not needed, the all-contributors plugin will add you automatically. I need to go back and convert these old ones

File: packages/gitbeaker-node/src/index.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ export const {
   Deployments,
   DeployKeys,
   Environments,
+  FreezePeriods,
```
Comment (jdalrymple) [inline packages/gitbeaker-node/src/index.ts:41]: Could you make this same change, to the gitbeaker-browser index file please

**Comment by jdalrymple** [comment]
Two small comments!

File: README.md (hunk 1 above)
Comment (nlochschmidt) [reply README.md]: Ok, done

**Comment by jdalrymple** [comment]
:rocket: PR was released in [`24.3.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/24.3.0) :rocket:

**Pull request merged**